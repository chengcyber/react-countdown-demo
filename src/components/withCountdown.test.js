import React from 'react'
import { mount, shallow } from 'enzyme'
import withCountDown from './withCountdown'

const mockTick = ({tick}) => <div>{tick.current}</div>

function setupMount() {
  const Component = withCountDown()(mockTick)

  const wrapper = mount(
    <Component />
  )

  return wrapper
}

function setupShallow() {
  const Component = withCountDown()(mockTick)

  const wrapper = shallow(
    <Component />
  )

  return wrapper
}

describe('with count down hoc', () => {

  it('should initial render correct', () => {
    const wrapper = setupShallow()
    expect(wrapper).toMatchSnapshot()
  })

  it('should count down working', () => {
    const wrapper = setupMount()
    const props = wrapper.find('mockTick').props()
    const div = wrapper.find('div')

    props.tick.start()
    expect(div.text()).toBe("59")
  })

})

