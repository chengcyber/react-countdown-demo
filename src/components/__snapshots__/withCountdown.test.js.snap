// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with count down hoc should initial render correct 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReactTimeout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "cancelAnimationFrame": [Function],
      "clearImmediate": [Function],
      "clearInterval": [Function],
      "clearTimeout": [Function],
      "requestAnimationFrame": [Function],
      "setImmediate": [Function],
      "setInterval": [Function],
      "setTimeout": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "cancelAnimationFrame": [Function],
        "clearImmediate": [Function],
        "clearInterval": [Function],
        "clearTimeout": [Function],
        "requestAnimationFrame": [Function],
        "setImmediate": [Function],
        "setInterval": [Function],
        "setTimeout": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
